src/components/Geolocation.react.js

```diff
@@ -0,0 +1,188 @@
+import {Component} from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * The CurrentLocation component gets geolocation of the device from the web browser.  See more info here:
+ * https://developer.mozilla.org/en-US/docs/Web/API/Geolocation_API
+ */
+
+export default class Geolocation extends Component {
+    constructor(props) {
+        super(props);
+        this.success = this.success.bind(this);
+        this.error = this.error.bind(this);
+        this.updatePosition = this.updatePosition.bind(this);
+    }
+
+    updatePosition() {
+        if (!navigator.geolocation) {
+            alert('Your browser does not support Geolocation');
+        } else {
+            this.props.setProps({
+                update_now: false,
+            });
+
+            const positionOptions = {
+                enableHighAccuracy: this.props.high_accuracy,
+                maximumAge: this.props.maximum_age,
+                timeout: this.props.timeout,
+            };
+
+            navigator.geolocation.getCurrentPosition(
+                this.success,
+                this.error,
+                positionOptions
+            );
+        }
+    }
+
+    componentDidMount() {
+        this.updatePosition();
+    }
+
+    componentDidUpdate(prevProps) {
+        if (
+            this.props.update_now ||
+            prevProps.maximum_age !== this.props.maximum_age ||
+            prevProps.timeout !== this.props.timeout ||
+            prevProps.high_accuracy !== this.props.high_accuracy
+        ) {
+            this.updatePosition();
+        }
+    }
+
+    success(pos) {
+        const crd = pos.coords;
+        const position_obj = {
+            lat: crd.latitude,
+            lon: crd.longitude,
+            accuracy: crd.accuracy,
+            alt: crd.altitude,
+            alt_accuracy: crd.altitudeAccuracy,
+            speed: crd.speed,
+            heading: crd.heading,
```

Comment:
Is there anything in `pos.coords` that we don't want in `position_obj`? If not, we can just use `pos.coords` directly. Unless the goal is to ensure that when there's always an entry for each item by the time this gets to Python, but if we pass an `undefined` to `JSON.stringify` it'll be deleted anyway, so if we want that we'll have to do something like `speed: crd.speed ?? null` (that's the [null coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator))

Reply:
I'd like to have everything from `pos.coords` available.  The only advantage I see of doing it this way is to rename the props.  `lat` and `lon` are much easier to type than `latitude` and `longitude`

Reply:
Ah yes, I didn’t notice the name changes, you have it right. Then perhaps just the `null` fallback for items that may be missing.

Reply:
Ok added the `null` fallback